dbms/src/Flash/Mpp/MPPTunnel.h

```diff
@@ -137,11 +159,14 @@ class TunnelSender : private boost::noncopyable
         std::shared_future<String> future;
         std::atomic<bool> msg_has_set{false};
     };
+
     MemoryTrackerPtr memory_tracker;
-    MPMCQueue<TrackedMppDataPacketPtr> send_queue;
     ConsumerState consumer_state;
     const LoggerPtr log;
     const String tunnel_id;
+    
+    std::atomic<size_t> * data_size_in_queue; // From MppTunnel
```

Comment:
I think `std::atomic<int64>` is better， since it may be lower than zero if some corner cases are encountered such as "sub early"

Reply:
> I think `std::atomic<int64>` is better， since it may be lower than zero if some corner cases are encountered such as "sub early"

okk